### ISSUE TITLE
IDE0060 is reported incorrectly when parameters has nullable attributes attached

### ISSUE THREAD

**Issue by marek-safar** [dotnet/roslyn-analyzers#6177]
### Analyzer

**Diagnostic ID**: IDE0060

### Analyzer source

**Version**: 7.0.0

### Describe the bug

IDE0060 message is issues for any partial method that uses nullable System.Diagnostics.CodeAnalysis attributes

### Steps To Reproduce

```c#
using System.Diagnostics.CodeAnalysis;

Class.Test();

partial class Class
{
    public static void Test ()
    {
        object o;
        M1 (ref o);
        M2 (ref o);
    }

    static partial void M1 ([NotNull] ref object i); // warning IDE0060: Remove unused parameter 'i'

    static partial void M2 (ref object i); // success
}
```

### Expected behavior

No warning to be reported

### Actual behavior

IDE0060 warning is reported for `M1` method
